### PR TITLE
More documentation

### PR DESCRIPTION
**Pull request by afs**
* Examples moved
* xloader linked in
* Query execution building and substitution

This aligns with PR apache/jena#1110.

### CODE REVIEW

File: source/documentation/sparql-apis/__index.md [hunk 1]
```diff
@@ -109,6 +107,41 @@ objects have been removed.
 `ModelStore` are the replacement for remote operations. `RDFConnection` and
 `RDFLink` provide APIs.
 
+## Substitution
+
+All query and update builders provide operations to uses a query and substitute
```
Comment (rvesse) [inline source/documentation/sparql-apis/__index.md:112]: Typo - `uses` -> `use`

File: source/documentation/tdb/faqs.md [hunk 2]
```diff
@@ -4,6 +4,8 @@ title: TDB FAQs
 
 ## FAQs
 
+
+-   [What are TDB1 and TDB2?](#tdv1-tdb2)
```
Comment (rvesse) [inline source/documentation/tdb/faqs.md:8]: Typo in anchor link `tdv1` -> `tdb1`

File: source/documentation/tdb/faqs.md [hunk 3]
```diff
@@ -77,11 +88,22 @@ As noted above to resolve this problem you **must** rebuild your database from t
 be repaired. This is why we **strongly** recommend you use [transactions](tdb_transactions.html) since this protects your dataset against 
 corruption.
 
+## What is `tdb.xloader`?
+
+`tdb1.xloader` and `tdb2.xloader` are bulk laodrs for very large dataset that
```
Comment (rvesse) [inline source/documentation/tdb/faqs.md:93]: Typo `laodrs` -> `loaders`

Also `dataset` should be `datasets`

File: source/documentation/tdb/tdb-xloader.md [hunk 4]
```diff
@@ -7,16 +7,19 @@ is stability and reliability for long running loading, running on modest and
 
 xloader is not a replacement for regular TDB1 and TDB2 loaders.
 
-"tdb1.xloader" was called "tdbloader2" and has some improvements.
+There are two scripts to load data using the xlaoder subsystem.
```
Comment (rvesse) [inline source/documentation/tdb/tdb-xloader.md:10]: Typo `xlaoder` -> `xloader`

File: source/documentation/tdb/tdb-xloader.md [hunk 5]
```diff
@@ -7,16 +7,19 @@ is stability and reliability for long running loading, running on modest and
...
+
+"tdb1.xloader", which was called "tdbloader2" and has some improvements.
 
 It is not as fast as other TDB loaders on dataset where the general loaders work
 on without encountering progressive slowdown.
 
-The xloaders for TDB1 and TDB2 are not identical. The TDB2 is more capable; it
-is based on the same design approach with further refinements to building the
-node table and to reduce the total amount of temporary file space used.
+The xloaders for TDB1 and TDB2 are not identical. The TDB2 xlaoder is more
```
Comment (rvesse) [inline source/documentation/tdb/tdb-xloader.md:17]: Typo `xlaoder` -> `xloader`

**Review by rvesse — APPROVED** [review]
Looks good, various small typos

File: source/documentation/sparql-apis/__index.md [hunk 6]
```diff
@@ -89,26 +89,59 @@ objects have been removed.
...
 * Deprecate modifying `QueryExecution` after it is built.
 
-* Parameterization for remote queries.
-  Parameterization - replacing variables by values before sending
-  a query - makes the query into a template. The same applies to updates.
-  This is also provided uniformly for local queries and should be used in
-  preference to the local-only "initial binding" approach which is
-  similarly but not identical.
+* Substitution of variables for concrete values in query and update execution.
+  This is a form of paramterization that works in both local and remnote usage
```
Comment (kinow) [inline source/documentation/sparql-apis/__index.md:93]: s/paramterization/parameterization (I think)

Comment (kinow) [reply source/documentation/sparql-apis/__index.md:93]: Also s/remnote/remote

File: source/documentation/sparql-apis/__index.md [hunk 7]
```diff
@@ -89,26 +89,59 @@ objects have been removed.
...
+  (unlike "intial bindings" which are only available for lcoal query execution).
```
Comment (kinow) [inline source/documentation/sparql-apis/__index.md:94]: s/lcoal/local

Comment (kinow) [reply source/documentation/sparql-apis/__index.md:94]: Also s/intial/initial

File: source/documentation/sparql-apis/__index.md [hunk 8]
```diff
@@ -89,26 +89,59 @@ objects have been removed.
...
+## Substitution
+
+All query and update builders provide operations to use a query and substitute
+variables for concrete RDF terms in the execution.
+
+Unlike "initial bindings" substitution is provided in query and update builders
+for both local and remote cases. 
+
+Substitution is always "replace variable with RDF term" in a query or update
+that is correct syntax. This means is does not apply to `INSERT DATA` or `DELETE
```
Comment (kinow) [inline source/documentation/sparql-apis/__index.md:119]: s/This means is/This means it

File: source/documentation/sparql-apis/__index.md [hunk 9]
```diff
@@ -89,26 +89,59 @@ objects have been removed.
...
+Substitution is to be preferred over "initial bindings" because it is clearly
+defined and applies to both query and update in both local and remote uses.
+
+"Substitution" and "initial bindings" are similar but not identical.
+
+See also 
+* [Parameterized Queries](documentation/query/parameterized-sparql-strings.html) 
+* [Jena Query Builder](https://jena.apache.org/documentation/extras/querybuilder/index.html)
+
+which provide a different ways to build a query.
```
Comment (kinow) [inline source/documentation/sparql-apis/__index.md:143]: s/a different ways/a different way (or /different ways/)

File: source/documentation/tdb/faqs.md [hunk 10]
```diff
@@ -18,6 +19,15 @@ title: TDB FAQs
 -   [What is the *Unable to check TDB lock owner, the lock file contents appear to be for a TDB2 database. Please try loading this location as a TDB2 database* error?](#tdb2-lock)
 -   [My question isn't answered here?](#not-answered)
 
+<a name="tdb1-tdb2></a>
+## TDB1 and TDB2
+
+TDB2 is a later generation of database for Jena. It is more robust and can
+handle large update transactions.
+
+These are different databases systems - the have different on-disk file formats
```
Comment (kinow) [inline source/documentation/tdb/faqs.md:28]: s/the have/they have ?

File: source/documentation/tdb/tdb-xloader.md [hunk 11]
```diff
@@ -3,20 +3,24 @@ title: TDB xloader
 ---
 
 TDB xloader ("x" for external) is a bulkloader for very large datasets. The goal
-is stability and reliability for long running loading, running on modest and
+is stability and reliability for long running loading, running on modest
+hardware and can load to storage of rotating disk or SSD.
```
Comment (kinow) [inline source/documentation/tdb/tdb-xloader.md:7]: I didn't understand the "can load to storage of rotating disk", but might be just lack of coffee :coffee: :laughing:

File: source/documentation/tdb/tdb-xloader.md [hunk 12]
```diff
@@ -30,22 +34,24 @@ or
 
 `tdb1.xloader --loc DIRECTORY` FILE...
 
-Additioally, there is an argument `--tmpdir` to use a different directory for
+Additionally, there is an argument `--tmpdir` to use a different directory for
 temporary files.
 
-`FILE` is any RDF syntax supported by Jena.
+`FILE` is any RDF syntax supported by Jena. Syntax is detemined by file
```
Comment (kinow) [inline source/documentation/tdb/tdb-xloader.md:40]: s/detemined/determined

File: source/documentation/tdb/tdb-xloader.md [hunk 13]
```diff
@@ -30,22 +34,24 @@ or
...
+extension and can include an addtional ".gz" or ".bz2" for compresses files.
```
Comment (kinow) [inline source/documentation/tdb/tdb-xloader.md:41]: s/compresses/compressed ?

File: source/documentation/tdb/tdb-xloader.md [hunk 14]
```diff
@@ -30,22 +34,24 @@ or
...
-`xloader` uses a lot of temporary disk space. 
-
 To avoid a load failing due to a syntax or other data error, it is advisable to
 run `riot --check` on the data first. Parsing is faster than loading.
 
-If desired, the data can be converted to [RDF Thrift](../io/rdf-binary.html) at
-this stage by adding `--stream rdf-thrift` to the riot checking run.
-Parsing RDF Thrift is faster than parsing N-Triples although the bulk of the loading process is not limited by parser speed.
+The TDB databases will take up a lot of disk space and in addition during
+loading `xloader` uses a significant amout of temporary disk space.
```
Comment (kinow) [inline source/documentation/tdb/tdb-xloader.md:49]: s/amout/amount

**Review by kinow — APPROVED** [review]
Found a few typos, but new docs look good! Thanks @afs

File: source/documentation/tdb/tdb-xloader.md (hunk 11 above)
Comment (afs) [reply source/documentation/tdb/tdb-xloader.md:7]: Or too much coffee (in my timezone)!

**Comment by afs** [comment]
@kinow  - thanks for the prompt review.

**Pull request merged**